### PR TITLE
Documentation for downstream TLS client auth

### PR DESCRIPTION
**Pull request by k-wall**
### Type of change

_Select the type of your PR_

- Documentation

### Description

#1631 added the ability to use TLS client auth on the downstream (client) side.  This PR adds documentation for that feature.

### Additional Context

_Why are you making this pull request?_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Update documentation
- [ ] Make sure all unit/integration tests pass
- [ ] Make sure all Sonarcloud warnings are addressed or are justifiably ignored.
- [ ] If applicable to the change, [trigger the system test suite](../blob/main/DEV_GUIDE.md#jenkins-pipeline-for-system-tests).  Make sure tests pass.
- [ ] If applicable to the change, [trigger the performance test suite](../blob/main/PERFORMANCE.md#jenkins-pipeline-for-performance). Ensure that any degradations to performance numbers are understood and justified.
- [ ] Ensure the PR references relevant issue(s) so they are closed on merging.
- [ ] For user facing changes, update CHANGELOG.md (remember to include changes affecting the API of the test artefacts too).

> **_NOTE:_**  You must be a member of `@kroxylicious/developers` to trigger the system test and performance test suites.  If you are not part of this group, comment on the PR requesting a trigger, tagging `@kroxylicious/developers`.

### CODE REVIEW

**Comment by k-wall** [comment]
@alanrobinson-dwp if you could review this too, that would be appeciated.

**Comment by alanrobinson-dwp** [comment]
> @alanrobinson-dwp if you could review this too, that would be appeciated.

Looks good to me!

File: docs/modules/configuring/con-configuring-client-connections.adoc [hunk 1]
```diff
@@ -11,6 +11,8 @@ This may require wildcard certificates and Subject Alternative Names (SANs).
 * Provide the TLS configuration using the tls properties in the virtual cluster's configuration to enable it to present the certificate to clients. 
 Depending on your certificate format, apply one of the following examples.
 
+You may also configure the virtual cluster to use TLS client authentication.
```
Comment (PaulRMellor) [inline docs/modules/configuring/con-configuring-client-connections.adoc:14]: Should we change the title of this section to "Securing client connections"

File: docs/modules/configuring/con-configuring-client-connections.adoc [hunk 2]
```diff
@@ -11,6 +11,8 @@ This may require wildcard certificates and Subject Alternative Names (SANs).
 * Provide the TLS configuration using the tls properties in the virtual cluster's configuration to enable it to present the certificate to clients.
```
Comment (PaulRMellor) [inline docs/modules/configuring/con-configuring-client-connections.adoc:11]:
```suggestion
* Provide the TLS configuration using the `tls` properties in the virtual cluster's configuration to enable it to present the certificate to clients. 
```

File: docs/modules/configuring/con-configuring-client-connections.adoc [hunk 3]
```diff
@@ -47,23 +49,39 @@ virtualClusters:
...
 <1> Private key of the virtual cluster.
-<2> Public CA certificate of the virtual cluster.
+<2> Public certificate of the virtual cluster.
 <3> (Optional) Password for the key.
 
-If required, configure the `insecure` property to disable trust and establish insecure connections with any Kafka Cluster, irrespective of certificate validity. 
-However, this option is only intended for use in development and testing environments where proper certificates are hard to obtain and manage.
+You may configure the virtual cluster to require that clients present a certificate to the virtual
+cluster. The virtual cluster verifies the client's certificate is signed by one of the CA certificates
+contained with a trust store.  If verification fails, the client's connection will be refused.
```
Comment (PaulRMellor) [inline docs/modules/configuring/con-configuring-client-connections.adoc:57]:
```suggestion
You can configure the virtual cluster to require that clients present a certificate for authentication. 
The virtual cluster verifies that the client's certificate is signed by one of the CA certificates contained in a trust store.  
If verification fails, the client's connection is refused.
```

File: docs/modules/configuring/con-configuring-client-connections.adoc [hunk 4]
```diff
@@ -47,23 +49,39 @@ virtualClusters:
...
+          clientAuth: REQUIRED # <4>
 # …
 ----
-<1> Enables insecure TLS.
+<1> PKCS #12 store containing CA certificate(s) used to verify that the client's certificate is trusted.
+<2> (Optional) Password to protect the PKCS #12 store.
+<3> (Optional) Keystore type. If a keystore type is not specified, the default JKS (Java Keystore) type is used.
+<4> If set to the value `REQUIRED`, the client must present a valid certificate.  If set to `REQUESTED` the client
+    will be requested to present a certificate.  If a certificate is presented, it will be validated. If the client
+    chooses not to present a certificate the connection is still allowed.  The value `NONE` disables client authentication.
```
Comment (PaulRMellor) [inline docs/modules/configuring/con-configuring-client-connections.adoc:81]:
```suggestion
<4> Client authentication mode. 
If set to `REQUIRED`, the client must present a valid certificate. 
If set to `REQUESTED`, the client is requested to present a certificate. If presented, the certificate is validated. If the client chooses not to present a certificate the connection is still allowed. 
If set to `NONE`, client authentication is disabled.
```

File: docs/modules/configuring/con-configuring-client-connections.adoc [hunk 5]
```diff
@@ -47,23 +49,39 @@ virtualClusters:
...
+
+NOTE: The client's identity established by TLS client authentication is not relayed to the target cluster. See
```
Comment (PaulRMellor) [inline docs/modules/configuring/con-configuring-client-connections.adoc:83]:
```suggestion
NOTE: The client's identity, as established through TLS client authentication, is currently not relayed to the target cluster. 
```

File: docs/modules/configuring/con-configuring-client-connections.adoc [hunk 6]
```diff
@@ -47,23 +49,39 @@ virtualClusters:
...
+(https://github.com/kroxylicious/kroxylicious/issues/1637)[issue].
```
Comment (PaulRMellor) [inline docs/modules/configuring/con-configuring-client-connections.adoc:84]:
```suggestion
For more information, see the https://github.com/kroxylicious/kroxylicious/issues/1637[related issue].
```

File: docs/modules/configuring/con-configuring-client-connections.adoc [hunk 7]
```diff
@@ -47,23 +49,39 @@ virtualClusters:
...
+
+
+
```
Comment (PaulRMellor) [inline docs/modules/configuring/con-configuring-client-connections.adoc:87]:
```suggestion
```

**Review by PaulRMellor — COMMENTED** [review]
Thanks @k-wall. I've left a few suggestions around wording.

File: docs/modules/configuring/con-configuring-client-connections.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (k-wall) [reply docs/modules/configuring/con-configuring-client-connections.adoc]: Done.

**Comment by k-wall** [comment]
Thanks for the feedback @PaulRMellor, I've incorporated your feedback.

**Review by SamBarker — APPROVED** [review]
LGTM 

Thanks @k-wall

**Pull request merged**